web/src/compose_actions.js

```diff
@@ -224,6 +224,10 @@ export function start(msg_type, opts) {
 
     opts = fill_in_opts_from_current_narrowed_view(msg_type, opts);
 
+    if (opts.trigger === "empty_narrow_compose") {
+        compose_recipient.set_selected_recipient_id("");
+    }
```

Comment:
Hmm, I worry that this will incorrectly cause us to fail to reply to the current stream when your current view is an empty narrow for a topic; can you explain the bug you're fixing here in more detail?

Reply:
It works fine since `compose_state.set_stream_id(opts.stream_id);` is being set later if there is a stream present.

The issue I am trying to fix is `recipient_id` being set in the current narrow for compose from previous narrow if the current narrow doesn't has a recipient like an empty search.

Reply:
Pushed to add that as a comment.